catalog-rest-service/src/main/java/org/openmetadata/catalog/resources/databases/TableResource.java

```diff
@@ -330,8 +332,11 @@ public Response patch(
           JsonPatch patch)
       throws IOException, ParseException {
     Fields fields = new Fields(FIELD_LIST, FIELDS);
+    List<MetadataOperation> metadataOperations = JsonPatchMetadataOperationMapper.getMetadataOperations(patch);
     Table table = dao.get(uriInfo, id, fields);
-    SecurityUtil.checkAdminRoleOrPermissions(authorizer, securityContext, dao.getOwnerReference(table));
+    SecurityUtil.checkAdminRoleOrPermissions(
+        authorizer, securityContext, dao.getEntityInterface(table).getEntityReference(), metadataOperations);
```

Comment:
entityReference might not be enough, as we need the ownership of the table to check if the logged-in user is the owner or belongs to a team thats the owner of the table

Reply:
I'm pushing this down into the Authorizer implementation:
- Authorizer gets entityReference of the entity
- Authorizer looks up owner of the entity using the entityReference and makes necessary decision

This way we centralize  all the entity related metadata gathering needed to make the decision within the Authorizer implementation.

Reply:
sounds good to me.